### PR TITLE
Add current sensor

### PR DESCRIPTION
**Pull request by theicfire**

### CODE REVIEW

**Comment by jpanikulam** [comment]
Awesome; Did you manage to get it working with the servo!?

File: embedded/current_sensor/ina219_test.cc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//%deps(ina219)
+
+#include "third_party/ina219/Adafruit_INA219.hh"
+#include <iostream>
+#include <unistd.h> // usleep
+
+int main() {
+    Adafruit_INA219 ina219 = Adafruit_INA219("/dev/i2c-1", 0x40);
```
Comment (jpanikulam) [inline embedded/current_sensor/ina219_test.cc]: [nit] Suggest `constexpr int this_is_what_the_hexnumber_means = 0x40`

File: embedded/current_sensor/ina219_test.cc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ina219.begin();
+    std::cout << "Current Sensor Test" << std::endl;
+    while (true) {
+        usleep(100 * 1000);
+        float shuntvoltage = 0;
+        float busvoltage = 0;
+        float current_mA = 0;
+        float loadvoltage = 0;
+        float power_mW = 0;
```
Comment (jpanikulam) [inline embedded/current_sensor/ina219_test.cc]: [nit] `const` these to be permitted to enter Valhalla

Comment (jpanikulam) [reply embedded/current_sensor/ina219_test.cc]: Ignore

File: embedded/current_sensor/ina219_test.cc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+        shuntvoltage = ina219.getShuntVoltage_mV();
+        busvoltage = ina219.getBusVoltage_V();
+        current_mA = ina219.getCurrent_mA();
+        power_mW = ina219.getPower_mW();
+        loadvoltage = busvoltage + (shuntvoltage / 1000);
```
Comment (jpanikulam) [inline embedded/current_sensor/ina219_test.cc]: The woke thing is to do 

`const float shuntvoltage = ..`

instead od pre-declaring at initializing to zero

**Comment by theicfire** [comment]
Yup, this does work with the servo! However, they're both separately opening up the i2c file, and I haven't tried out multiple current sensors (which will all open up an i2c file....)

So I need to have both the servo and the current monitor share the i2c file. I suggest putting that in a separate PR though

File: embedded/current_sensor/ina219_test.cc (hunk 3 above)
Comment (theicfire) [reply embedded/current_sensor/ina219_test.cc]: Yup, that was just me copying Adafruit. Fixed.

File: third_party/ina219/Adafruit_INA219.hh [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+  void init();
+  void wireWriteRegister(uint8_t reg, uint16_t value);
+  void wireReadRegister(uint8_t reg, uint16_t *value);
+  int16_t getBusVoltage_raw(void);
+  int16_t getShuntVoltage_raw(void);
+  int16_t getCurrent_raw(void);
+  int16_t getPower_raw(void);
+  const char *_i2cBus;
+uint8_t _address;
```
Comment (jpanikulam) [inline third_party/ina219/Adafruit_INA219.hh]: Clang format this in your next PR

**Pull request merged**